File: db/mime_type.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in mime_type.
*/

class MIME_Type extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "mime_type";
    $this->order = "";
    $this->domain = "valuelist";
    $this->field['mime_type_id']['type'] = 'SERIAL';
    $this->field['mime_type_id']['not_null'] = true;
    $this->field['mime_type']['type'] = 'VARCHAR';
    $this->field['mime_type']['length'] = 128;
    $this->field['mime_type']['not_null'] = true;
    $this->field['file_extension']['type'] = 'VARCHAR';
    $this->field['file_extension']['length'] = 16;
    $this->field['f_image']['type'] = 'BOOL';
    $this->field['f_image']['not_null'] = true;
    $this->field['f_image']['default'] = true;
    $this->field['mime_type_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/mime_type.rb
module Momomoto
  class Mime_type < Base
    def initialize
      super
      @domain = 'valuelist'
      @fields = {
        :mime_type_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :mime_type => Datatype::Varchar.new( {:not_null=>true, :length=>128} ),
        :file_extension => Datatype::Varchar.new( {:length=>16} ),
        :f_image => Datatype::Bool.new( {:not_null=>true, :default=>true} )
      }
    end
  end
end
